apps/meteor/client/views/room/providers/SelectedMessagesProvider.tsx

```diff
@@ -14,8 +16,19 @@ export const selectedMessageStore = new (class SelectMessageStore extends Emitte
 > {
 	store = new Set<string>();
 
+	private storeArray = Array.from(this.store);
+
+	availableMessages = new Set<string>();
+
 	isSelecting = false;
 
+	constructor() {
+		super();
+		this.on('change', () => {
+			this.storeArray = Array.from(this.store);
+		});
```

Comment:
Why are you saving the set as an array in memory at all times? When selecting a large number of messages (toggle all for example) this would run many times and can be expensive depending on the ammount of messages already selected.

Reply:
removed!